src/views/Home.jsx

```diff
@@ -1,7 +1,41 @@
 import './Home.css';
 import { SingleList } from '../components/SingleList';
+import { useEffect, useState } from 'react';
+import { createList } from '../api/firebase';
+import { useNavigate } from 'react-router-dom';
+
+export function Home({ data, setListPath, user }) {
+	const [shoppingListName, setShoppingListName] = useState('');
+	const navigate = useNavigate();
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		const { uid, email } = user;
+
+		const names = data.map((list) => {
+			return list.name;
+		});
+
+		if (names.includes(shoppingListName)) {
+			alert('name should be unique');
+
+			return;
+		}
+
+		createList(uid, email, shoppingListName);
+		alert('created');
+		navigate('/list');
+	};
+
+	useEffect(() => {
+		const newestList = data[data.length - 1];
```

Comment:
curious - do we know that lists are always returned in the order they were created?